Signal/src/Jobs/MarkIdentityAsSeenJob.swift

```diff
@@ -8,18 +8,24 @@ import Foundation
 class MarkIdentityAsSeenJob: NSObject {
     let TAG = "[MarkIdentityAsSeenJob]"
 
-    private let thread: TSThread
+    private let recipientIds: [String]
 
     public class func run(thread: TSThread) {
-        MarkIdentityAsSeenJob(thread: thread).run()
+        let recipientIds = thread.recipientIdentifiers
+
+        MarkIdentityAsSeenJob(recipientIds: recipientIds).run()
+    }
+
+    public class func run(recipientId: String) {
+        MarkIdentityAsSeenJob(recipientIds: [recipientId]).run()
     }
 
-    init(thread: TSThread) {
-        self.thread = thread
+    init(recipientIds: [String]) {
+        self.recipientIds = recipientIds
     }
 
     public func run() {
-        for recipientId in self.thread.recipientIdentifiers {
+        for recipientId in self.recipientIds {
```

Comment:
As written this job does run sync to allow inline "confirm and continue" type behavior. This job potentially entails some db writes, shouldn't take too long.

Maybe some kind of naming convention like `run`/`runAsync` is sufficient? If you agree I'll go through the existing jobs to make sure that convention is honored.

WDYT @charlesmchen?